spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/EmbeddedCassandra.scala

```diff
@@ -16,48 +16,146 @@ trait EmbeddedCassandra {
+    * @param configTemplates name of the cassandra.yaml template resources
     * @param forceReload if set to true, the server will be reloaded fresh even if the configuration didn't change */
-  def useCassandraConfig(configTemplate: String, forceReload: Boolean = false) {
+  def useCassandraConfig(configTemplates: Seq[String], forceReload: Boolean = false) {
     import EmbeddedCassandra._
+    if (!hosts.isEmpty && configTemplates.size > hosts.size) {
+      throw new IllegalArgumentException("Configuration templates can't be more than the number of specified hosts")
+    }
     if (sys.env.get(HostProperty).isEmpty) {
-      if (currentConfigTemplate != configTemplate || forceReload) {
-        clearCache()
-        cassandra.map(_.destroy())
-        cassandra = Some(new CassandraRunner(configTemplate))
-        currentConfigTemplate = configTemplate
+      clearCache()
+
+      for (i <- configTemplates.indices) {
+        if (configTemplates(i) == null || configTemplates(i).trim.isEmpty) {
+          throw new IllegalArgumentException("Configuration template can't be null or empty")
+        }
+        if (currentConfigTemplates.lift(i) != Option(configTemplates(i)) || forceReload) {
+          cassandras.lift(i).map(_.map(_.destroy()))
+          cassandras = cassandras.patch(i, Seq(Some(new CassandraRunner(configTemplates(i), getProps(i)))), 1)
+          currentConfigTemplates = currentConfigTemplates.patch(i, Seq(configTemplates(i)), 1)
+        }
       }
     }
   }
 }
 
+
 object EmbeddedCassandra {
 
-  val DefaultHost = "127.0.0.1"
+  val HostProperty = "IT_TEST_CASSANDRA_HOSTS"
+  val NativePortProperty = "IT_TEST_CASSANDRA_NATIVE_PORTS"
+  val RpcPortProperty = "IT_TEST_CASSANDRA_RPC_PORTS"
+
+  validate
+
+  private def validate() = {
+    val hosts = sys.env.get(HostProperty)
+    if (hosts != None) {
+      val nativePorts = sys.env.get(NativePortProperty)
+      if (nativePorts != None) {
+        val rpcPorts = sys.env.get(RpcPortProperty)
+        if (rpcPorts != None) {
+          val hostSize = hosts.getOrElse("").split(",").size
+          val nativePortSize = nativePorts.getOrElse("").split(",").size
+          if (hostSize != nativePortSize) {
+            throw new RuntimeException("IT_TEST_CASSANDRA_HOSTS must have same size as IT_TEST_CASSANDRA_NATIVE_PORTS")
+          }
+          val rpcPortSize = rpcPorts.getOrElse("").split(",").size
+          if (hostSize != rpcPortSize) {
+            throw new RuntimeException("IT_TEST_CASSANDRA_HOSTS must have same size as IT_TEST_CASSANDRA_RPC_PORTS")
+          }
+        } else {
+          throw new RuntimeException("Missing IT_TEST_CASSANDRA_RPC_PORTS settings in system environment")
+        }
+      } else {
+        throw new RuntimeException("Missing IT_TEST_CASSANDRA_NATIVE_PORTS settings in system environment")
+      }
+    }
+  }
 
-  // TODO change to CASSANDRA_HOST
-  val HostProperty = "IT_TEST_CASSANDRA_HOST"
+  val hosts: IndexedSeq[InetAddress] = {
```

Comment:
Maybe enclose all the settings provided externally in a case class or something?